### ISSUE TITLE
Audio output always goes through speakers even if headphones are connected

### ISSUE THREAD

**Issue by handlerug** [mpv-android/mpv-android#217]
It's worth noting that I also hear system sounds (touch, screen lock, etc.) always through speakers too, but all media apps I know except of MPV-Android use headphones (YouTube, Spotify, VLC, MX Player, etc.). I guess this problem occurs because MPV handles all audio and video itself and doesn't properly route it through Android sound channels.

I have Android 9 (Resurrection Remix OS 7.0.1 ROM from openkirin.net), Honor 9 STF-AL10. I've never experienced such problem on EMUI (stock ROM for Huawei/Honor devices) and on OmniROM (custom ROM).

**Comment by sfan5** [comment]
mpv-android uses OpenSLES for audio output, most other media apps probably use the Java AudioTrack API (see also #135).
The system sound things sounds like an oddity with your custom ROM.

**Comment by handlerug** [comment]
Looks like it's related to this particular ROM, yeah. I tried to reproduce those problems on another device and everything was fine. Maybe OpenSL ES output isn't rerouted to default Android media channel output device (or it outputs to system sounds channel by default)? 

P.S. I tried reproducing this issue on my current device again, and some mystery is going on there. Both MPV (which uses OpenSL ES API as you said), VLC (which can use this API too, I selected it in settings) **and** system sounds are broken and have some strange behaviour, part of which is described in my first comment. So it definitely looks like ROM problem. Thanks for the response!
